python/pyarrow/tests/test_serialization.py

```diff
@@ -297,6 +297,27 @@ def test_numpy_immutable(large_memory_map):
             result[0] = 1.0
 
 
+# see https://issues.apache.org/jira/browse/ARROW-1695
+def test_serialization_callback_numpy():
+
+    class DummyClass(object):
+        pass
+
+    def serialize_dummy_class(obj):
+        x = np.zeros(4)
+        return x
+
+    def deserialize_dummy_class(serialized_obj):
+        return serialized_obj
+
+    pa._default_serialization_context.register_type(
+        DummyClass, "DummyClass", pickle=False,
+        custom_serializer=serialize_dummy_class,
+        custom_deserializer=deserialize_dummy_class)
+
+    pa.serialize(DummyClass())
```

Comment:
Should also deserialize the object and assert that it is the correct value.

Reply:
Oops too late. This is a regression test to catch the segfault which used to already occur in the serialize call, so only testing that is fine; the actual codepath of serializing a numpy array via custom serializer will be tested in the PR I'm about to create, which is about serializing pytorch Tensors.

Reply:
You want to handle pytorch tensors in Arrow? As opposed to in Ray?

Reply:
Yes, I'd like to handle them in Arrow conditionally on pytorch being installed, much in the same way as we handle pandas. There is precedent for that, the glib bindings do it for raw lua torch tensors I think.

Reply:
Ok, sounds good.